transaction/core/src/amount/proof_of_opening.rs

```diff
@@ -0,0 +1,286 @@
+//! Code related to zero knowledge "proof of opening" for a Pedersen Commitment
+//!
+//! TODO: Add appropriate references
+//! (Page 27): https://www.cs.purdue.edu/homes/ninghui/courses/555_Spring12/handouts/555_Spring12_topic23.pdf
+
+use crate::{
+    domain_separators::{PROOF_OF_OPENING_PUBLIC_DOMAIN_TAG, PROOF_OF_OPENING_SECRET_DOMAIN_TAG},
+    ring_signature::{CurveScalar, PedersenGens},
+    Commitment,
+};
+use core::convert::TryFrom;
+use curve25519_dalek::{
+    ristretto::{CompressedRistretto, RistrettoPoint},
+    scalar::Scalar,
+};
+use mc_crypto_digestible::{Digestible, MerlinTranscript};
+use mc_crypto_keys::CompressedRistrettoPublic;
+use prost::Message;
+use serde::{Deserialize, Serialize};
+
+/// This object is the transcript of a non-interactive "proof of opening"
```

Comment:
Reading through the slides, is this better described as a proof of openability?

Reply:
Yeah I'm open to other names for this thing

I think "proof of opening" jibes with other things I've read?

We could also call it like "proof of generator" or "proof of pedersen base" or something like this? That's sort of more to the point, it proves that the prover built this commitment over a particular set of pedersen generators?

Reply:
The best description I have so far is "a proof of knowledge of a Pedersen commitment opening". That terminology is used here, https://eprint.iacr.org/2020/061.pdf

So, ProofOfOpening seems ok to me.